pkg/sources/git/git.go

```diff
@@ -45,9 +47,6 @@ type Source struct {
 	sources.Progress
 	conn        *sourcespb.Git
 	scanOptions *ScanOptions
-	// Kludge to preserve engine.ScanGit functionality which doesn't expect
-	// the scanning to clean up the directory.
-	preserveTempDirs bool
```

Comment:
Is this removed because we never need to do it anymore?

Reply:
We do need to still do it, it's just now it's handled within the source [here](https://github.com/trufflesecurity/trufflehog/blob/dca78d25650e3c5d07350440f655426dbe7b0812/pkg/sources/git/git.go#L286-L286). We had the kludge in place b/c the caller in `main` used to handle the cleanup, but since i removed all that logic in main and in the engine, the source can correctly handle it.

Reply:
This should also hopefully more closely align with all the other functionality where the source is responsible for cleaning up any temp dirs.

Reply:
oh, by "this" i meant the kludge. i should have been more clear!

Reply:
Right we don't need to do this anymore. It was there for just to preserve the weird behavior we had with the engine and the source. The source takes care of that functionality now.

Reply:
Thanks for cleaning up my kludge. I'm happy to see it go!